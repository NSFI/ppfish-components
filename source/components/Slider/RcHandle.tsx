import React from 'react';
import classNames from 'classnames';
import { addEventListener } from '../../utils';

interface HandleProps {
  prefixCls?: string;
  className?: string;
  vertical?: boolean;
  offset?: number;
  style?: React.CSSProperties;
  disabled?: boolean;
  min?: number;
  max?: number;
  value?: number;
  tabIndex?: number;
  handle?: React.ReactNode;
  onMouseEnter?: React.MouseEventHandler;
  onMouseLeave?: React.MouseEventHandler;
}

interface HandleStates {
  clickFocused: boolean;
}

export default class Handle extends React.Component<HandleProps, HandleStates> {
  state = {
    clickFocused: false,
  };

  private onMouseUpListener: any;

  private handle: any;

  componentDidMount() {
    // mouseup won't trigger if mouse moved out of handle,
    // so we listen on document here.
    this.onMouseUpListener = addEventListener(document, 'mouseup', this.handleMouseUp);
  }

  componentWillUnmount() {
    if (this.onMouseUpListener) {
      this.onMouseUpListener.remove();
    }
  }

  setClickFocus(focused) {
    this.setState({ clickFocused: focused });
  }

  handleMouseUp = () => {
    if (document.activeElement === this.handle) {
      this.setClickFocus(true);
    }
  };

  handleBlur = () => {
    this.setClickFocus(false);
  };

  handleKeyDown = () => {
    this.setClickFocus(false);
  };

  clickFocus() {
    this.setClickFocus(true);
    this.focus();
  }

  focus() {
    this.handle.focus();
  }

  blur() {
    this.handle.blur();
  }

  render() {
    const {
      prefixCls,
      vertical,
      offset,
      style,
      disabled,
      min,
      max,
      value,
      tabIndex,
      handle,
      ...restProps
    } = this.props;

    const elClassName = classNames({
      [this.props.className]: true,
      [`${prefixCls}-handle-custom`]: !!this.props.handle,
      [`${prefixCls}-handle-click-focused`]: this.state.clickFocused,
    });

    const postionStyle = vertical ? { bottom: `${offset}%` } : { left: `${offset}%` };
    const elStyle = {
      ...style,
      ...postionStyle,
    };
    let ariaProps = {};
    if (value !== undefined) {
      ariaProps = {
        ...ariaProps,
        'aria-valuemin': min,
        'aria-valuemax': max,
        'aria-valuenow': value,
        'aria-disabled': !!disabled,
      };
    }

    return (
      <div
        ref={node => (this.handle = node)}
        role="slider"
        tabIndex={disabled ? null : tabIndex || 0}
        {...ariaProps}
        {...restProps}
        className={elClassName}
        style={elStyle}
        onBlur={this.handleBlur}
        onKeyDown={this.handleKeyDown}
      >
        {!!handle && handle}
      </div>
    );
  }
}
