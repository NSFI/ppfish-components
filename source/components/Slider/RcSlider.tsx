/* eslint-disable react/prop-types */
import React from 'react';
import warning from 'warning';
import Track from './common/Track';
import createSlider from './common/createSlider';
import * as utils from './utils';
import { polyfill } from 'react-lifecycles-compat';
import type { GenericSliderProps, GenericSliderState } from './interface';

export interface SliderProps extends GenericSliderProps {
  value?: number;
  defaultValue?: number;
  autoFocus?: boolean;
  min?: number;
  max?: number;
  step?: number | null;
  prefixCls?: string;
  onChange?: (value: number) => void;
  onBeforeChange?: (value: number) => void;
  onAfterChange?: (value: number) => void;
  vertical?: boolean;
  included?: boolean;
  disabled?: boolean;
  reverse?: boolean;
  minimumTrackStyle?: React.CSSProperties;
  trackStyle?: React.CSSProperties;
  handleStyle?: React.CSSProperties;
  tabIndex?: number;
  startPoint?: number;
  handle?: (props: {
    className: string;
    prefixCls?: string;
    vertical?: boolean;
    offset: number;
    value: number;
    dragging?: boolean;
    disabled?: boolean;
    min?: number;
    max?: number;
    reverse?: boolean;
    index: number;
    tabIndex?: number;
    style?: React.CSSProperties;
    ref?: React.Ref<any>;
  }) => React.ReactElement;
}

interface SliderStates extends GenericSliderState {
  value: number;
  dragging: boolean;
}
class Slider extends React.Component<SliderProps, SliderStates> {
  static getDerivedStateFromProps(nextProps, prevState) {
    if (!('value' in nextProps || 'min' in nextProps || 'max' in nextProps)) return null;

    const prevValue = prevState.value;
    const value = nextProps.value !== undefined ? nextProps.value : prevValue;
    const val = utils.ensureValueInRange(value, nextProps);
    const nextValue = utils.ensureValuePrecision(val, nextProps);
    if (nextValue === prevValue) return null;

    return { value: nextValue };
  }

  private startValue: any;

  private startPosition: any;

  private prevMovedHandleIndex: any;

  private removeDocumentEvents: any;

  private saveHandle: any;

  private calcValueByPos: any;

  private focus: any;

  private calcOffset: any;

  constructor(props) {
    super(props);

    const defaultValue = props.defaultValue !== undefined ? props.defaultValue : props.min;
    const value = props.value !== undefined ? props.value : defaultValue;

    this.state = {
      value: this.trimAlignValue(value),
      dragging: false,
    };
    if (process.env.NODE_ENV !== 'production') {
      warning(
        !('minimumTrackStyle' in props),
        'minimumTrackStyle will be deprecate, please use trackStyle instead.',
      );
      warning(
        !('maximumTrackStyle' in props),
        'maximumTrackStyle will be deprecate, please use railStyle instead.',
      );
    }
  }

  componentDidMount() {
    const { autoFocus, disabled } = this.props;
    if (autoFocus && !disabled) {
      this.focus();
    }
  }

  componentDidUpdate(prevProps, prevState) {
    let prevValue = prevState.value;
    let currentValue = this.state.value;
    const { onChange } = this.props;
    if (prevValue !== currentValue) {
      if (
        onChange &&
        utils.isValueOutOfRange(currentValue, {
          min: this.props.min,
          max: this.props.max,
        })
      ) {
        onChange(currentValue);
      }
    }
  }

  onChange(state) {
    const props = this.props;
    const isNotControlled = !('value' in props);
    if (isNotControlled) {
      this.setState(state);
    }

    const changedValue = state.value;
    props.onChange(changedValue);
  }

  onStart(position) {
    this.setState({ dragging: true });
    const props = this.props;
    const prevValue = this.getValue();
    props.onBeforeChange(prevValue);

    const value = this.calcValueByPos(position);
    this.startValue = value;
    this.startPosition = position;

    if (value === prevValue) return;

    this.prevMovedHandleIndex = 0;

    this.onChange({ value });
  }

  onEnd = () => {
    this.setState({ dragging: false });
    this.removeDocumentEvents();
    this.props.onAfterChange(this.getValue());
  };

  onMove(e, position) {
    utils.pauseEvent(e);
    const { value: oldValue } = this.state;
    const value = this.calcValueByPos(position);
    if (value === oldValue) return;

    this.onChange({ value });
  }

  onKeyboard(e) {
    const valueMutator = utils.getKeyboardValueMutator(e);

    if (valueMutator) {
      utils.pauseEvent(e);
      const state = this.state;
      const oldValue = state.value;
      const mutatedValue = valueMutator(oldValue, this.props);
      const value = this.trimAlignValue(mutatedValue);
      if (value === oldValue) return;

      this.onChange({ value });
    }
  }

  getValue() {
    return this.state.value;
  }

  getLowerBound() {
    return this.props.min;
  }

  getUpperBound() {
    return this.state.value;
  }

  trimAlignValue(v, nextProps = {}) {
    const mergedProps = { ...this.props, ...nextProps };
    const val = utils.ensureValueInRange(v, {
      min: mergedProps.min,
      max: mergedProps.max,
    });
    return utils.ensureValuePrecision(val, mergedProps);
  }

  render() {
    const {
      prefixCls,
      vertical,
      included,
      disabled,
      minimumTrackStyle,
      trackStyle,
      handleStyle,
      tabIndex,
      min,
      max,
      handle: handleGenerator,
    } = this.props;
    const { value, dragging } = this.state;
    const offset = this.calcOffset(value);
    const handle = handleGenerator({
      className: `${prefixCls}-handle`,
      prefixCls,
      vertical,
      offset,
      value,
      dragging,
      disabled,
      min,
      max,
      index: 0,
      tabIndex,
      style: handleStyle[0] || handleStyle,
      ref: h => this.saveHandle(0, h),
    });

    const _trackStyle = trackStyle[0] || trackStyle;
    const track = (
      <Track
        className={`${prefixCls}-track`}
        vertical={vertical}
        included={included}
        offset={0}
        length={offset}
        style={{
          ...minimumTrackStyle,
          ..._trackStyle,
        }}
      />
    );

    return { tracks: track, handles: handle };
  }
}

polyfill(Slider);
export default createSlider(Slider);
