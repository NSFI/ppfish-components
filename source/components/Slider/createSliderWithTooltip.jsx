import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '../Tooltip/index.tsx';
import Handle from './RcHandle';

export default function createSliderWithTooltip(Component) {
  return class ComponentWrapper extends React.Component {
    static propTypes = {
      handleStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.arrayOf(PropTypes.object)]),
      tipProps: PropTypes.object,
    };
    static defaultProps = {
      handleStyle: [{}],
      tipProps: {},
    };
    constructor(props) {
      super(props);
      this.state = { visibles: {} };
    }
    handleTooltipVisibleChange = (index, visible) => {
      this.setState((prevState) => {
        return {
          visibles: {
            ...prevState.visibles,
            [index]: visible,
          },
        };
      });
    }
    handleWithTooltip = ({ value, dragging, index, disabled, ...restProps }) => {
      const {
        tipProps,
        handleStyle,
      } = this.props;

      const {
        prefixCls = 'rc-slider-tooltip',
        placement = 'top',
        visible = visible || false,
        ...restTooltipProps,
      } = tipProps;

      let handleStyleWithIndex;
      if (Array.isArray(handleStyle)) {
        handleStyleWithIndex = handleStyle[index] || handleStyle[0];
      } else {
        handleStyleWithIndex = handleStyle;
      }

      return (
        <Tooltip
          {...restTooltipProps}
          prefixCls={prefixCls}
          placement={placement}
          visible={(!disabled && (this.state.visibles[index] || dragging)) || visible}
          key={index}
        >

          <Handle
            {...restProps}
            style={{
              ...handleStyleWithIndex,
            }}
            value={value}
            onMouseEnter={() => this.handleTooltipVisibleChange(index, true)}
            onMouseLeave={() => this.handleTooltipVisibleChange(index, false)}
          />
        </Tooltip>
      );
    }
    render() {
      return <Component {...this.props} handle={this.handleWithTooltip} />;
    }
  };
}
