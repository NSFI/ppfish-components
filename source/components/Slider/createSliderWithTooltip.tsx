import React from 'react';
import Tooltip, { TooltipProps } from '../Tooltip/index';
import Handle from './RcHandle';

interface WrapperProps {
  handleStyle: React.CSSProperties[] | React.CSSProperties;
  tipProps: TooltipProps;
}

interface WrapperStates {
  visibles: {
    [index: number]: boolean;
  };
}

export default function createSliderWithTooltip(Component) {
  return class ComponentWrapper extends React.Component<WrapperProps, WrapperStates> {
    static defaultProps = {
      handleStyle: [{}],
      tipProps: {},
    };

    constructor(props) {
      super(props);
      this.state = { visibles: {} };
    }

    handleTooltipVisibleChange = (index, visible) => {
      this.setState(prevState => {
        return {
          visibles: {
            ...prevState.visibles,
            [index]: visible,
          },
        };
      });
    };

    handleWithTooltip = ({ value, dragging, index, disabled, ...restProps }) => {
      const { tipProps, handleStyle } = this.props;

      const {
        prefixCls = 'rc-slider-tooltip',
        placement = 'top',
        visible = false,
        ...restTooltipProps
      } = tipProps;

      let handleStyleWithIndex;
      if (Array.isArray(handleStyle)) {
        handleStyleWithIndex = handleStyle[index] || handleStyle[0];
      } else {
        handleStyleWithIndex = handleStyle;
      }

      return (
        <Tooltip
          {...restTooltipProps}
          prefixCls={prefixCls}
          placement={placement}
          visible={(!disabled && (this.state.visibles[index] || dragging)) || visible}
          key={index}
        >
          <Handle
            {...restProps}
            style={{
              ...handleStyleWithIndex,
            }}
            value={value}
            onMouseEnter={() => this.handleTooltipVisibleChange(index, true)}
            onMouseLeave={() => this.handleTooltipVisibleChange(index, false)}
          />
        </Tooltip>
      );
    };

    render() {
      return <Component {...this.props} handle={this.handleWithTooltip} />;
    }
  };
}
