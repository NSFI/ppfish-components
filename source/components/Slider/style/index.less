@import "../../../assets/css/index";

@prefixClass: ~"@{fishd-prefix}-slider";

@ease-out-quint      : cubic-bezier(0.23, 1, 0.32, 1);
@ease-in-quint       : cubic-bezier(0.755, 0.05, 0.855, 0.06);

.borderBox() {
  box-sizing: border-box;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0); //  remove tap highlight color for mobile safari

  * {
    box-sizing: border-box;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0); //  remove tap highlight color for mobile safari
  }
}

.@{prefixClass} {
  .reset-component;
  position: relative;
  margin: @slider-margin;
  padding: 4px 0;
  height: 12px;
  cursor: pointer;
  .borderBox();

  &-with-marks {
    margin-bottom: 28px;
  }

  &-rail {
    position: absolute;
    width: 100%;
    height: 4px;
    border-radius: @slider-border-radius-base;
    background-color: @slider-rail-background-color;
    transition: background-color .3s;
  }

  &-track {
    position: absolute;
    height: 4px;
    border-radius: @slider-border-radius-base;
    background-color: @slider-track-background-color;
    transition: background-color 0.3s ease;
  }

  &-handle {
    position: absolute;
    margin-left: -7px;
    margin-top: -5px;
    width: 14px;
    height: 14px;
    cursor: pointer;
    cursor: -webkit-grab;
    cursor: grab;
    border-radius: 50%;
    border: solid 2px @slider-handle-color;
    background-color: @component-background;
    transition: border-color .3s, transform .3s cubic-bezier(0.18, 0.89, 0.32, 1.28);
    touch-action: pan-x;

    &:focus {
      border-color: @slider-handle-color-focus;
      outline: none;
    }

    // ??
    &-click-focused:focus {
      border-color: tint(@primary-color, 50%);
      box-shadow: unset;
    }

    &.@{fishd-prefix}-tooltip-open {
      border-color: @slider-handle-color-tooltip-open;
    }
  }

  &:hover {
    .@{prefixClass}-rail {
      background-color: @slider-rail-background-color-hover;
    }
    .@{prefixClass}-track {
      background-color: @slider-track-background-color-hover;
    }
    .@{prefixClass}-handle:not(.@{fishd-prefix}-tooltip-open) {
      border-color: @slider-handle-color-hover;
    }
  }

  &-mark {
    position: absolute;
    top: 14px;
    left: 0;
    width: 100%;
    font-size: 14px;
  }

  &-mark-text {
    position: absolute;
    display: inline-block;
    vertical-align: middle;
    text-align: center;
    cursor: pointer;
    color: @text-color-tertiary;

    &-active {
      color: @text-color-secondary;
    }
  }

  &-step {
    position: absolute;
    width: 100%;
    height: 4px;
    background: transparent;
  }

  &-dot {
    position: absolute;
    bottom: -2px;
    margin-left: -4px;
    width: 8px;
    height: 8px;
    border: 2px solid @slider-dot-border-color;
    background-color: @component-background;
    cursor: pointer;
    border-radius: 50%;
    vertical-align: middle;
    &:first-child {
      margin-left: -4px;
    }
    &:last-child {
      margin-left: -4px;
    }
    &-active {
      border-color: @slider-dot-border-color-active;
    }
  }

  &-disabled {
    cursor: not-allowed;

    .@{prefixClass}-track {
      background-color: @slider-disabled-color !important;
    }

    .@{prefixClass}-handle,
    .@{prefixClass}-dot {
      border-color: @slider-disabled-color !important;
      background-color: @component-background;
      cursor: not-allowed;
      box-shadow: none;
    }

    .@{prefixClass}-mark-text,
    .@{prefixClass}-dot {
      cursor: not-allowed!important;
    }
  }
}

.@{prefixClass}-vertical {
  width: 12px;
  height: 100%;
  margin: 6px 10px;
  padding: 0 4px;

  .@{prefixClass} {
    &-rail {
      height: 100%;
      width: 4px;
    }

    &-track {
      width: 4px;
    }

    &-handle {
      margin-left: -5px;
      margin-bottom: -7px;
      touch-action: pan-y;
    }

    &-mark {
      top: 0;
      left: 12px;
      width: 18px;
      height: 100%;
    }

    &-mark-text {
      left: 4px;
      white-space: nowrap;
    }

    &-step {
      width: 4px;
      height: 100%;
    }

    &-dot {
      top: auto;
      left: 2px;
      margin-bottom: -4px;
      &:first-child {
        margin-bottom: -4px;
      }
      &:last-child {
        margin-bottom: -4px;
      }
    }
  }
}
