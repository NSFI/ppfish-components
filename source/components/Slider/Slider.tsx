import * as React from 'react';
import RcSlider from './RcSlider';
import RcRange from './RcRange';
import RcHandle from './RcHandle';
import Tooltip from '../Tooltip';

import './style/index.less';

export interface SliderMarks {
  [key: number]: React.ReactNode | {
    style: React.CSSProperties,
    label: React.ReactNode,
  };
}

export type SliderValue = number | [number, number];

export type HandleGeneratorFn = (info: {
  value: number,
  dragging: boolean,
  index: number,
  rest: any[],
}) => React.ReactElement<any>;

export interface SliderProps {
  prefixCls?: string;
  tooltipPrefixCls?: string;
  range?: boolean;
  min?: number;
  max?: number;
  step?: number | null;
  marks?: SliderMarks;
  dots?: boolean;
  value?: SliderValue;
  defaultValue?: SliderValue;
  included?: boolean;
  disabled?: boolean;
  vertical?: boolean;
  onChange?: (value: SliderValue) => void;
  onAfterChange?: (value: SliderValue) => void;
  tipFormatter?: null | ((value: number) => React.ReactNode);
  className?: string;
  id?: string;
}

export interface SliderState {
  visibles: { [index: number]: boolean };
}

export default class Slider extends React.Component<SliderProps, SliderState> {
  static defaultProps = {
    prefixCls: 'fishd-slider',
    tooltipPrefixCls: 'fishd-tooltip',
    tipFormatter(value: number) {
      return value.toString();
    },
  };

  private rcSlider: any;

  constructor(props: SliderProps) {
    super(props);
    this.state = {
      visibles: {},
    };
  }

  toggleTooltipVisible = (index: number, visible: boolean) => {
    this.setState(({ visibles }) => ({
      visibles: {
        ...visibles,
        [index]: visible,
      },
    }));
  }

  handleWithTooltip: HandleGeneratorFn = ({ value, dragging, index, ...restProps }) => {
    const { tooltipPrefixCls, tipFormatter } = this.props;
    const { visibles } = this.state;
    const visible = tipFormatter ? (visibles[index] || dragging)  : false;
    return (
      <Tooltip
        prefixCls={tooltipPrefixCls}
        title={tipFormatter ? tipFormatter(value) : ''}
        visible={visible}
        placement="top"
        transitionName="zoom-down"
        key={index}
      >
        <RcHandle
          {...restProps}
          value={value}
          onMouseEnter={() => this.toggleTooltipVisible(index, true)}
          onMouseLeave={() => this.toggleTooltipVisible(index, false)}
        />
      </Tooltip>
    );
  }

  focus() {
    this.rcSlider.focus();
  }

  blur() {
    this.rcSlider.focus();
  }

  saveSlider = (node: any) => {
    this.rcSlider = node;
  }

  render() {
    const { range, ...restProps } = this.props;
    if (range) {
      return <RcRange {...restProps} ref={this.saveSlider} handle={this.handleWithTooltip} />;
    }
    return <RcSlider {...restProps} ref={this.saveSlider} handle={this.handleWithTooltip} />;
  }
}
