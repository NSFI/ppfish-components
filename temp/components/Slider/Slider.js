var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import * as React from 'react';
import RcSlider from './RcSlider';
import RcRange from './RcRange';
import RcHandle from './RcHandle';
import Tooltip from '../Tooltip';
var Slider = /** @class */ (function (_super) {
    __extends(Slider, _super);
    function Slider(props) {
        var _this = _super.call(this, props) || this;
        _this.toggleTooltipVisible = function (index, visible) {
            _this.setState(function (_a) {
                var _b;
                var visibles = _a.visibles;
                return ({
                    visibles: __assign(__assign({}, visibles), (_b = {}, _b[index] = visible, _b))
                });
            });
        };
        _this.sliderHandle = function (_a) {
            var value = _a.value, dragging = _a.dragging, index = _a.index, restProps = __rest(_a, ["value", "dragging", "index"]);
            var _b = _this.props, tooltipPrefixCls = _b.tooltipPrefixCls, tipFormatter = _b.tipFormatter, _c = _b.tipMode, tipMode = _c === void 0 ? 'default' : _c, handle = _b.handle;
            var visibles = _this.state.visibles;
            var visible = tipFormatter ? visibles[index] || dragging : false;
            if (tipMode === 'all') {
                return React.createElement(RcHandle, __assign({}, restProps, { handle: handle, value: value }));
            }
            return (React.createElement(Tooltip, { prefixCls: tooltipPrefixCls, title: tipFormatter ? tipFormatter(value) : '', visible: visible, placement: "top", transitionName: "zoom-down", key: index, stretch: "minWidth" },
                React.createElement(RcHandle, __assign({}, restProps, { value: value, handle: handle, onMouseEnter: function () { return _this.toggleTooltipVisible(index, true); }, onMouseLeave: function () { return _this.toggleTooltipVisible(index, false); } }))));
        };
        _this.saveSlider = function (node) {
            _this.rcSlider = node;
        };
        _this.state = {
            visibles: {}
        };
        return _this;
    }
    Slider.prototype.focus = function () {
        this.rcSlider.focus();
    };
    Slider.prototype.blur = function () {
        this.rcSlider.focus();
    };
    Slider.prototype.render = function () {
        var _a = this.props, range = _a.range, restProps = __rest(_a, ["range"]);
        if (range) {
            return React.createElement(RcRange, __assign({}, restProps, { ref: this.saveSlider, handle: this.sliderHandle }));
        }
        return React.createElement(RcSlider, __assign({}, restProps, { ref: this.saveSlider, handle: this.sliderHandle }));
    };
    Slider.defaultProps = {
        prefixCls: 'fishd-slider',
        tooltipPrefixCls: 'fishd-tooltip',
        tipFormatter: function (value) {
            return value.toString();
        }
    };
    return Slider;
}(React.Component));
export default Slider;
