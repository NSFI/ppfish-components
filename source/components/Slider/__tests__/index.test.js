import React from 'react';
import { render, mount } from 'enzyme';
import Slider from '..';

describe('Slider', () => {
  it('should show tooltip when hovering slider handler', () => {
    const wrapper = mount(<Slider defaultValue={30} />);
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseEnter');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseLeave');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });

  it('should show correct placement tooltip when set tooltipPlacement', () => {
    const wrapper = mount(<Slider vertical defaultValue={30} tooltipPlacement="left" />);
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseEnter');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseLeave');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });

  it('when mouseEnter, tooltip should show', () => {
    let wrapper = mount(<Slider defaultValue={30} />);
    expect(wrapper.find('.fishd-tooltip').length).toBe(0);
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseEnter');
    expect(wrapper.find('.fishd-tooltip').at(0)).not.toBe(null);
    expect(wrapper.find('.fishd-tooltip').at(0).hasClass('fishd-tooltip-hidden')).toBe(false);
    wrapper.find('.fishd-slider-handle').at(0).simulate('click');
    expect(wrapper.find('.fishd-tooltip').at(0).hasClass('fishd-tooltip-hidden')).toBe(false);
  });

  it('when tipFormatter is null, tooltip should be hidden', () => {
    let wrapper = mount(<Slider defaultValue={30} tipFormatter={null} />);
    wrapper.find('.fishd-slider-handle').at(0).simulate('click');
    expect(wrapper.find('.fishd-tooltip').length).toBe(0);
  });

  it('when tipFormatter is not null, tooltip should show', () => {
    let wrapper = mount(<Slider defaultValue={30} tipFormatter={value => value.toString()} />);
    wrapper.find('.fishd-slider-handle').at(0).simulate('mouseEnter');
    expect(wrapper.find('.fishd-tooltip').at(0).hasClass('fishd-tooltip-hidden')).toBe(false);
  });

  it('when step is null, thumb can only be slided to the specific mark', () => {
    const intentionallyWrongValue = 40;
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(
      <Slider marks={marks} defaultValue={intentionallyWrongValue} step={null} />,
    );
    expect(wrapper.find('.fishd-slider-handle').get(0).props).toHaveProperty('value', 48);
  });

  it('when step is not null, thumb can be slided to the multiples of step', () => {
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(<Slider marks={marks} defaultValue={49} step={1} />);
    expect(wrapper.find('.fishd-slider-handle').get(0).props).toHaveProperty('value', 49);
  });

  it('when step is undefined, thumb can be slided to the multiples of step', () => {
    const marks = {
      0: '0',
      48: '48',
      100: '100',
    };

    const wrapper = mount(<Slider marks={marks} defaultValue={49} step={undefined} />);
    expect(wrapper.find('.fishd-slider-handle').get(0).props).toHaveProperty('value', 49);
  });

  it('tipFormatter should not crash with undefined value', () => {
    [undefined, null].forEach(value => {
      mount(<Slider value={value} />);
    });
  });
  it('step should not crash with undefined value', () => {
    [undefined, null].forEach(value => {
      mount(<Slider step={value} />);
    });
  });
});
