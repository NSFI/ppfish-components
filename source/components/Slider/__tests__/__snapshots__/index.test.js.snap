// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should show correct placement tooltip when set tooltipPlacement 1`] = `
<div>
  <div
    class="fishd-tooltip"
    style="opacity:0;pointer-events:none"
  >
    <div
      class="fishd-tooltip-content"
    >
      <div
        class="fishd-tooltip-arrow"
      />
      <div
        class="fishd-tooltip-inner"
      >
        30
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider should show correct placement tooltip when set tooltipPlacement 2`] = `<div />`;

exports[`Slider should show tooltip when hovering slider handler 1`] = `
<div>
  <div
    class="fishd-tooltip"
    style="opacity:0;pointer-events:none"
  >
    <div
      class="fishd-tooltip-content"
    >
      <div
        class="fishd-tooltip-arrow"
      />
      <div
        class="fishd-tooltip-inner"
      >
        30
      </div>
    </div>
  </div>
</div>
`;

exports[`Slider should show tooltip when hovering slider handler 2`] = `<div />`;
